pkg/operator/admission/controller.go

```diff
@@ -0,0 +1,118 @@
+/*
```

Comment:
should we create our own admission controller and server?
i would suggest using [webhook](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/webhook) package from `controller-runtime` considering we also (will) use `controller-runtime` in operator side.

Reply:
yeah @vbnrh, this is only a suggestion. feel free to go ahead with your current approach :grin:

Reply:
Hey @prksu, It was a good suggestion worth looking into. Thanks for it.

Maybe we'll have some follow up on this in future..